python/python/lance/dataset.py

```diff
@@ -689,6 +689,83 @@ def cleanup_old_versions(
             td_to_micros(older_than), delete_unverified
         )
 
+    def create_scalar_index(
+        self,
+        column: Union[str, List[str]],
+        index_type: str,
+        name: Optional[str] = None,
+        replace: bool = True,
```

Comment:
nit: I think boolean options should almost always be keyword only, for the sake of readability.
```suggestion
        *,
        replace: bool = True,
```

Reply:
Good idea.  I've changed this to keyword only